packages/base-driver/lib/protocol/protocol.js

```diff
@@ -239,9 +239,17 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
       // this command should never be proxied (which is useful for plugin developers who add
       // commands and generally would not want that command to be proxied instead of handled by the
       // plugin)
+      let pluginOverrodeProxy = false;
       if (isSessCmd && !spec.neverProxy && driverShouldDoJwpProxy(driver, req, spec.command)) {
-        await doJwpProxy(driver, req, res);
-        return;
+        if (!driver.pluginsToHandleCmd ||
```

Comment:
I wonder if `pluginsToHandleCmd` should be a noop by default?  I do not love with all the checking-for-existence -of-functions which seems to happen often 😛

Reply:
then this would break when used with a driver that doesn't rely on the latest version of basedriver. so we'll be stuck with the check anyway if we want compatibility with older drivers.

Reply:
oh, also, i don't really want the concept of plugins to apply to drivers other than the umbrella appium driver. it's sort of a special case. i don't like that we have to mention it in the protocol at all, there's probably some massive rearchitecture that's possible to get proxying and plugins working together more generically.

all that to say, i only want `driver.pluginsToHandleCmd` to exist when the protocol is working with `AppiumDriver`, not any other driver. if i trusted instanceof checks on custom classes I guess we could do it that way

Reply:
ok, I'll ask more about this later